deploy/docker-swarm/README.md

```diff
@@ -0,0 +1,97 @@
+# Docker Swarm Hetzner CSI plugin
+
+Currently in Beta. Please consult the Docker Swarm documentation
+for cluster volumes (=CSI) support at https://github.com/moby/moby/blob/master/docs/cluster_volumes.md
+
+## How to install the plugin
+
+Run the following steps on all nodes (especially master nodes).
+The simplest way to achieve this
+
+1. Install the plugin 
+
+```bash
+docker plugin install --disable --alias hetznercloud/csi-driver-docker --grant-all-permissions hetznercloud/csi-driver-docker
+```
+
+2. Set HCLOUD_TOKEN
+
+```bash
+docker plugin set hetznercloud/csi-driver-docker HCLOUD_TOKEN=<your token>
+```
+
+3. Enable plugin
+
+```bash
+docker plugin enable hetznercloud/csi-driver-docker
+```
+
+## How to create a volume
+
+Example: Create a volume wih size 50G in Nuremberg:
+
+```bash
+docker volume create --driver hetznercloud/csi-driver-docker --required-bytes 50G --type mount --sharing onewriter --scope single hcloud-debug1 --topology-required nbg1
+```
+
+We can now use this in a service:
+
+```bash
+docker service create --name hcloud-debug-serv1   --mount type=cluster,src=hcloud-debug1,dst=/srv/www   nginx:alpine
+```
+
+Note that only scope `single` is supported as Hetzner Cloud volumes can only be attached to one node at a time
+
+We can however share the volume on multiple containers on the same host:
+
+```bash
+docker volume create --driver hetznercloud/csi-driver-docker --required-bytes 50G --type mount --sharing all --scope single hcloud-debug1 --topology-required nbg1
+```
+
+After creation we can now use this volume with `--sharing all` in more than one replica:
+
+```bash
+docker service create --name hcloud-debug-serv2  --mount type=cluster,src=hcloud-debug2,dst=/srv/www   nginx:alpine
+docker service scale hcloud-debug-serv2=2
+```
+
+## How to resize a docker swarm Hetzner CSI volume
+
+Currently, the Docker Swarm CSI support does not come with support for volume resizing.
```

Comment:
Is there an upstream ticket that we can link here so users can check if support has been added by the time they read this guide?

Reply:
I just created one https://github.com/moby/moby/issues/44985

Reply:
done